sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java

```diff
@@ -371,54 +374,108 @@ public DataLakeServiceVersion getServiceVersion() {
+        String owner) {
         try {
-            return setAccessControlWithResponse(accessControl, null).flatMap(FluxUtil::toMono);
+            return setAccessControlListWithResponse(accessControlList, group, owner, null).flatMap(FluxUtil::toMono);
         } catch (RuntimeException ex) {
             return monoError(logger, ex);
         }
+    }
 
+    /**
+     * Changes the access control list for a resource.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setAccessControlListWithResponse#List-String-String-DataLakeRequestConditions}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
+     *
+     * @param accessControlList The access control list of the resource.
+     * @param group The group of the resource.
+     * @param owner The owner of the resource.
+     * @param accessConditions {@link DataLakeRequestConditions}
+     * @return A reactive response containing the resource info.
+     */
+    public Mono<Response<PathInfo>> setAccessControlListWithResponse(List<PathAccessControlEntry> accessControlList,
+        String group, String owner, DataLakeRequestConditions accessConditions) {
+        try {
+            return withContext(context -> setAccessControlWithResponse(accessControlList, null, group, owner,
+                accessConditions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
     }
 
     /**
-     * Changes the access control for a resource.
+     * Changes the permissions for a resource.
      *
      * <p><strong>Code Samples</strong></p>
      *
-     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setAccessControlWithResponse#PathAccessControl-DataLakeRequestConditions}
+     * {@codesnippet com.azure.storage.file.datalake.DataLakePathAsyncClient.setPermissions#PathPermissions-String-String}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure Docs</a></p>
      *
-     * @param accessControl {@link PathAccessControl}
+     * @param permissions The permissions of the resource.
+     * @param group The group of the resource.
+     * @param owner The owner of the resource.
+     * @return A reactive response containing the resource info.
+     */
+    public Mono<PathInfo> setPermissions(PathPermissions permissions, String group, String owner) {
+        try {
+            return setPermissionsWithResponse(permissions, group, owner, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Changes the permissions for a resource.
```

Comment:
For this and setAccessControlList, can you also mention that the api can be used to update the owner or owning group.

Reply:
done